index.rb

```diff
@@ -1 +1,139 @@
-#start
+class Station
+  attr_accessor :passenger_train
+
+  def initialize(station_name)
+    @station_name = station_name
+    @cargo_train = []
+    @passenger_train = []
+  end
+
+  def add_train(train)
+    if train.type == "passenger"
+      @passenger_train.push(train.num)
+    elsif train.type == "cargo"
+      @cargo_train.push(train.num)
+    end
+  end
+
+  def delete_train(train)
+    if train.type == "passenger"
+      @passenger_train.delete(train.num)
+    elsif train.type == "cargo"
+      @cargo_train.delete(train.num)
+    end
+    puts "Со станции отбыл поезд: #{train.num}"
+  end
+end
+
+class Route
+  attr_accessor :route
+
+  def initialize(start, ending)
+    @route =[start, ending]
+  end
+
+  def add_station(station)
+    @route.insert(-2, station)
+    puts "В маршрут была добавлена станция #{station}"
+  end
+
+  def delete_station(station)
+    if station == @route[0] || station == @route.last

```

Comment:
можно написать методы для первой и последней станции
тогда здесь запись будет читаться проще:
```suggestion
    if station == first_station || station == last_station

```